src/wasm/compilation.rs

```diff
@@ -147,7 +147,19 @@ impl Compilation {
     pub fn emit(mut self, module: Module, data_initializers: Vec<DataInitializer>) -> Result<Ref<CodeRef>> {
         self.relocate(&module)?;
 
-        let start_index = module.start_func?;
+        let start_index;

```

Comment:
You could rearrange this to be:
```rust
let start_index = if let Some(index) = module.start_func {
    index
} else if let Some(Export::Function(index)) = module.exports.get("main") {
    index
} else {
    Err(internal_error!())
}?;
```

Reply:
Well

```
let start_index = if let Some(index) = module.start_func {
    Ok(index)
} else if let Some(Export::Function(index)) = module.exports.get("main") {
    Ok(index)
} else {
    Err(internal_error!())
}?;
```

Stylistically I prefer the way I have it now, it doesn't hide the immediate return in a `?` flag, it doesn't create a result just to destruct it immediately. But I definitely don't feel strongly about it. If you are writing a lot of code the other way for instance I'd favor changing this to match existing style.

Reply:
If you prefer the original way, that's cool with me.